classes/security/UserGroupDAO.inc.php

```diff
@@ -779,46 +776,76 @@ function deleteSettingsByLocale($locale) {
 	 */
 	function _getSearchSql($searchType, $search, $searchMatch, &$params) {
 		$searchTypeMap = [
-			IDENTITY_SETTING_GIVENNAME => 'usgs.setting_value',
-			IDENTITY_SETTING_FAMILYNAME => 'usfs.setting_value',
-			USER_FIELD_USERNAME => 'u.username',
-			USER_FIELD_EMAIL => 'u.email',
-			USER_FIELD_AFFILIATION => 'us.setting_value',
+			IDENTITY_SETTING_GIVENNAME => "EXISTS(
+				SELECT 0
+				FROM user_settings usgs
+				WHERE usgs.user_id = u.user_id
+					AND usgs.setting_name = 'givenName'
+					AND LOWER(usgs.setting_value) LIKE LOWER(?)
+			)",
+			IDENTITY_SETTING_FAMILYNAME => "EXISTS(
+				SELECT 0
+				FROM user_settings usgs
+				WHERE usgs.user_id = u.user_id
+					AND usgs.setting_name = 'familyName'
+					AND LOWER(usgs.setting_value) LIKE LOWER(?)
+			)",
+			USER_FIELD_USERNAME => 'LOWER(u.username) LIKE LOWER(?)',
+			USER_FIELD_EMAIL => 'LOWER(u.email) LIKE LOWER(?)',
+			USER_FIELD_AFFILIATION => "EXISTS(
+				SELECT 0
+				FROM user_settings usgs
+				WHERE usgs.user_id = u.user_id
+					AND usgs.setting_name = 'affiliation'
+					AND LOWER(usgs.setting_value) LIKE LOWER(?)
+			)",
 		];
 
 		$searchSql = '';
-
+		$search = trim($search);
 		if (!empty($search)) {
-
 			if (!isset($searchTypeMap[$searchType])) {
-				$str = $this->concat('COALESCE(usgs.setting_value,\'\')', 'COALESCE(usfs.setting_value,\'\')', 'u.email', 'COALESCE(us.setting_value,\'\')');
-				$concatFields = ' ( LOWER(' . $str . ') LIKE ? OR LOWER(cves.setting_value) LIKE ? ) ';
-
-				$search = strtolower($search);
-
-				$words = preg_split('{\s+}', $search);
-				$searchFieldMap = array();
-
+				$userFilter = ' (' . $this->concat('LOWER(u.email)', 'LOWER(u.username)') . ' LIKE LOWER(?))';
+				$usersFilter = $settingsFilter = $interestsFilter = [];
+				$words = PKPString::regexp_split('/\s+/', $search);
 				foreach ($words as $word) {
-					$searchFieldMap[] = $concatFields;
+					$usersFilter[] = $userFilter;
+					$settingsFilter[] = 'LOWER(us.setting_value) LIKE LOWER(?)';
+					$interestsFilter[] = 'LOWER(cves.setting_value) LIKE LOWER(?)';
 					$term = '%' . $word . '%';
-					array_push($params, $term, $term);
+					array_push($params, $term, $term, $term);
 				}
 
-				$searchSql .= ' AND (  ' . join(' AND ', $searchFieldMap) . '  ) ';
+				$searchSql .= 'AND (
+					' . implode(' AND ', $usersFilter) . '
+					OR EXISTS(
+						SELECT 0
+						FROM user_settings us
+						WHERE
+							u.user_id = us.user_id
+							AND (' . implode(' AND ', $settingsFilter) . ')
+					)
```

Comment:
There were many joins on the `user_settings` just to check if there's a match, so I reduced to a single sub-query.

But the code isn't the same, here I'm checking for any user setting, while the previous code is searching just in the given name, family name and affiliation.